src/main/java/appeng/fluids/parts/PartFluidImportBus.java

```diff
@@ -100,28 +102,36 @@ protected TickRateModulation doBusWork()
 		}
 
 		final TileEntity te = this.getConnectedTE();
+
 		if( te != null && te.hasCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() ) )
 		{
 			try
 			{
 				final IFluidHandler fh = te.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() );
 				final IMEMonitor<IAEFluidStack> inv = this.getProxy().getStorage().getInventory( this.getChannel() );
+
 				if( fh != null )
 				{
-					FluidStack fluidStack = fh.drain( this.calculateAmountToSend(), false );
+					final FluidStack fluidStack = fh.drain( this.calculateAmountToSend(), false );
+
 					if( this.filterEnabled() && !this.isInFilter( fluidStack ) )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)